### PR TITLE
axisbelow should just set zorder.

### PR DESCRIPTION
**Pull request by anntzer**
See discussion starting at https://gitter.im/matplotlib/matplotlib?at=59a0b3b9bc46472974c206bf.

Please vote as to whether we should just deprecate the old axisbelow functions in favor of just using zorder everywhere (I think we should).

xref #5045 #6287

attn @afvincent 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by afvincent** [comment]
FWIW, the float option for `axisbelow` seems to already have been evaluated in a not too old past here #6287

**Comment by anntzer** [comment]
Actually I decided to not support float values in axisbelow -- just set zorder if that's what you want to do (in fact, one should just set zorder in all cases).

**Comment by anntzer** [comment]
It appears that commenting out the lines
```
            if not np.isfinite(posx) or not np.isfinite(posy):
                raise ValueError("posx and posy should be finite values")
```
in text.py makes the currently failing docs example (custom_scale_example) just build fine... strange.

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -2632,6 +2618,8 @@ def set_frame_on(self, b):
         Parameters
         ----------
         b : bool
+            ..
+                ACCEPTS: bool
```
Comment (NelleV) [inline lib/matplotlib/axes/_base.py]: Why did you add this line?

Comment (anntzer) [reply lib/matplotlib/axes/_base.py]: ... because the PR dates from before we got rid of them :)
Removing it.

**Comment by NelleV** [comment]
Apart from minor documentation problems, this looks good on the mpl side of things. I'm trying to stay away from the 3d stuff, so I can't comment on that.
Can I push directly to your branch to fix the couple of doc problems?

**Comment by NelleV** [comment]
:champagne:
